src/app/services/code-view/angular-code-service.ts

```diff
@@ -372,4 +374,25 @@ export class AngularCodeService extends CodeService {
         fileInput.appendTo(form)
         return form;
     }
-}
\ No newline at end of file
+
+    private checkForDvSample(demoFileMetadataName: string, demosBaseUrl: string): string {
+        let hasDvSample = false;
+        this.dvSamplesPaths.forEach(p => {
+            if (demoFileMetadataName.includes(p)) {
+                demoFileMetadataName = demoFileMetadataName.replace(p, "");
+                hasDvSample = true;
+            }
+        });
+
+        if (!hasDvSample) {
+            demoFileMetadataName = demoFileMetadataName.replace("/", "--");
+        }
+
+        let demoFileMetadataPath = `${demosBaseUrl}${this.demoFilesFolderUrlPath}${demoFileMetadataName}.json`;
```

Comment:
Why not introduce a new variable , similar to `this.demoFilesFolderUrlPath` - it will hold the path for DV samples:

![image](https://user-images.githubusercontent.com/14248932/184593336-3ed64dbe-aed0-4866-8e64-61dd3993f23b.png)

Then we wont need to work with strings here/